activation/post.go

```diff
@@ -23,13 +25,59 @@ type PostSetupProvider initialization.Provider
 
 // PostConfig is the configuration of the Post protocol, used for data creation, proofs generation and validation.
 type PostConfig struct {
-	MinNumUnits     uint32 `mapstructure:"post-min-numunits"`
-	MaxNumUnits     uint32 `mapstructure:"post-max-numunits"`
-	LabelsPerUnit   uint64 `mapstructure:"post-labels-per-unit"`
-	K1              uint32 `mapstructure:"post-k1"`
-	K2              uint32 `mapstructure:"post-k2"`
-	K3              uint32 `mapstructure:"post-k3"`
-	K2PowDifficulty uint64 `mapstructure:"post-k2pow-difficulty"`
+	MinNumUnits     uint32        `mapstructure:"post-min-numunits"`
+	MaxNumUnits     uint32        `mapstructure:"post-max-numunits"`
+	LabelsPerUnit   uint64        `mapstructure:"post-labels-per-unit"`
+	K1              uint32        `mapstructure:"post-k1"`
+	K2              uint32        `mapstructure:"post-k2"`
+	K3              uint32        `mapstructure:"post-k3"`
+	K2PowDifficulty uint64        `mapstructure:"post-k2pow-difficulty"`
+	PowDifficulty   PowDifficulty `mapstructure:"post-pow-difficulty"`
+}
+
+func (c PostConfig) ToConfig() config.Config {
+	return config.Config{
+		MinNumUnits:     c.MinNumUnits,
+		MaxNumUnits:     c.MaxNumUnits,
+		LabelsPerUnit:   c.LabelsPerUnit,
+		K1:              c.K1,
+		K2:              c.K2,
+		K3:              c.K3,
+		K2PowDifficulty: c.K2PowDifficulty,
+		PowDifficulty:   [32]byte(c.PowDifficulty),
+	}
+}
+
+type PowDifficulty [32]byte
+
+func (d PowDifficulty) String() string {
+	return fmt.Sprintf("%X", d[:])
+}
+
+func (d *PowDifficulty) UnmarshalText(text []byte) error {
+	decodedLen := hex.DecodedLen(len(text))
+	if decodedLen != 32 {
+		return fmt.Errorf("expected 32 bytes, got %d", decodedLen)
+	}
+	_, err := hex.Decode(d[:], text)
+	return err
+}
+
+// Decodes a hex string representation of PoW difficulty into a [32]byte.
+func DecodePowDifficulty(f reflect.Type, t reflect.Type, data any) (any, error) {
```

Comment:
I don't think you need this hook if `PowDifficulty` implements `TextUnmarshaler`. Instead you should be able to use `mapstructure.TextUnmarshallerHookFunc()`.